### PR TITLE
test(relation): simplifies tests for belongsTo relationship

### PR DESCRIPTION
**Pull request by stalniy**

### CODE REVIEW

File: test/unit/decorators/belongsTo.test.js [hunk 1]
```diff
@@ -1,184 +1,158 @@
...
+      const Foo = store.getMapper('foo').recordClass
+      const Bar = store.getMapper('bar').recordClass
+
+      assert.isTrue(Foo.prototype.hasOwnProperty('bars'))
+      assert.isTrue(Bar.prototype.hasOwnProperty('foo'))
+    })
+
+    it('should automatically map relations when record is added to store', function () {
+      assert.deepEqual(foos[0].bars, bars)
+      assert.isTrue(bars.every(bar => bar.foo === foos[0]))
```
Comment (jmdobry) [inline test/unit/decorators/belongsTo.test.js]: Does Babel transpile uses of `Array.prototype.every` into ES5-compatible code? If not then we can't use it.

Comment (stalniy) [inline test/unit/decorators/belongsTo.test.js]: The only browser which doesn't support `Array#every` is IE8. Currently a lot of libraries drop support of old browsers (Angular2, Ioinic2, Twitter Bootstrap 4). So, frankly speaking I don't believe this is an issue. And Node.js is currently at 6.9.1 which supports a lot more than just `Array#every`

Also this is in tests, but if you want to proceed with ES5 way anyway, I would like to add polyfill for tests like core.js. So, at least in tests people shouldn't care about ES5.

Does it work for you?

**Comment by stalniy** [comment]
I agree about `strictEqual` will double check and replace `deepEqual` with `strictEqual

File: test/unit/decorators/belongsTo.test.js (hunk 1 above)
Comment (stalniy) [inline test/unit/decorators/belongsTo.test.js]: If so, then we can't garranty that the library works under IE8, and its  the only browser which doesn't support some/every array methods. 

Then the reasonable question: does it make sense to worry about it?
People can just include es5 polyfill for older browsers like ie8 and that's it (e.g., using conditional comments) 
I can mention that in the docs if you want.

Comment (jmdobry) [inline test/unit/decorators/belongsTo.test.js]: Here is where the docs talk about this: http://www.js-data.io/v3.0/docs/installation#dependencies

**Comment by stalniy** [comment]
@jmdobry replaced `deepEqual` with `sameMembers` http://chaijs.com/api/assert/#method_samemembers

**Pull request merged**